data-model/src/main/java/io/micronaut/data/model/query/builder/AbstractSqlLikeQueryBuilder.java

```diff
@@ -1803,7 +1803,8 @@ public QueryResult buildOrderBy(String query, @NonNull PersistentEntity entity,
             List<Association> associations = new ArrayList<>(path.getAssociations());
             int assocCount = associations.size();
             // If last association is embedded, it does not need to be joined to the alias since it will be in the destination table
-            if (assocCount > 0 && associations.get(assocCount - 1) instanceof Embedded) {
+            // JPA/Hibernate is special case and in that case we leave association for specific handling below
```

Comment:
For JPA we shouldn't handle like for JDBC, didn't know how to recognize case except using `computePropertyPaths()` as it is being done [below too](https://github.com/micronaut-projects/micronaut-data/blob/master/data-model/src/main/java/io/micronaut/data/model/query/builder/AbstractSqlLikeQueryBuilder.java#L1942)